### PR TITLE
Split the fetch/init steps in bitbake scripts

### PR DESCRIPTION
**Pull request by tobsan**
It is not obvious that we always want to do fetchall.

### CODE REVIEW

File: yocto/initialize-bitbake.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+# WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR
+# BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES
+# OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+# WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+# ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+# SOFTWARE.
+#
+# For further information see LICENSE
+#
```
Comment (frznlogic) [inline yocto/initialize-bitbake.sh]: could we please have some documentation of the input arguments?

Comment (tobsan) [reply yocto/initialize-bitbake.sh]: Done.

File: yocto/fetch-sources-for-recipes.sh [hunk 2]
```diff
@@ -18,23 +18,28 @@
...
 # Fetch all sources
 echo "Fetching all the sources. Will try $COUNTER times in case of a bad connection."
 I=0
-while [  $I -lt $COUNTER ]; do
+while [ $I -lt $COUNTER ]; do
     echo "Try number: $I"
     let I+=1
     sleep 1
-    time bitbake -c fetchall $IMAGES
+    time bitbake -k -c fetchall $IMAGES
```
Comment (rpannek) [inline yocto/fetch-sources-for-recipes.sh]: So the `-k` will make it fail later? Isn't it good to know early that one source is not available and the image will not be able to be build as early as possible? Would would be the purpouse of downloading the rest for almost one hour if you won't be able to build anyway?

Comment (tobsan) [reply yocto/fetch-sources-for-recipes.sh]: In this script, even if fetchall fails, we retry 100 times, so failing late makes sense from the perspective of getting as many sources as possible per retry, plus the delay might make some hickup-y server get well between tries if we're lucky, but I see your point as well, and there is merit to both views. Would you like me to change it back?

Comment (rpannek) [reply yocto/fetch-sources-for-recipes.sh]: Good point with the hick up, I don't really have a preference.

**Pull request merged**